docs/architecture/decisions/0004-use-aws-load-balancer-controller-for-edge-traffic-services.md

```diff
@@ -0,0 +1,35 @@
+# 4. Use AWS Load Balancer Controller for edge traffic services
+
+Date: 2021-08-12
+
+## Status
+
+Accepted
+
+## Context
+
+We require a method of managing and direcing external internet traffic into the cluster. Kubernetes provides [several options for handling inbound traffic](https://medium.com/google-cloud/kubernetes-nodeport-vs-loadbalancer-vs-ingress-when-should-i-use-what-922f010849e0).
+
+We do not want to expose the cluster directly to the internet, and so require an AWS load balancer in front of the cluster. The load balancer must support TLS termination and integration with our DNS provider (AWS Route 53).
```

Comment:
I see you've already mentioned WAF and Shield below. We're actually using both already in production, so could potentially also mention them here as another reason for wanting an AWS load balancer in front of the cluster.

Reply:
Thanks, added a note on current WAF/Shield usage.